### ISSUE TITLE
why is it spewing multiple log lines during catchup?

### ISSUE THREAD

**Issue by jedmccaleb** [stellar/stellar-core#870]
This is during catchup obviously.

```
2015-10-21T15:02:49.958 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:49.966 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:49.988 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:49.999 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.010 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.018 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.028 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.037 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.047 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.056 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.065 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.076 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.085 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.093 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.101 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.109 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.120 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.128 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.138 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.146 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.154 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.163 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.173 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.183 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.220 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.258 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.273 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.290 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.321 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.338 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.354 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.371 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.389 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.408 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.427 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.450 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.468 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.482 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.507 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.524 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.540 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.559 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.569 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.578 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.587 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.595 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.607 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.627 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.636 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.645 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.653 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.670 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.678 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.686 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.695 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 218
2015-10-21T15:02:50.696 ca025c [] [History] WARN  Snapshot queued while already publishing
2015-10-21T15:02:50.707 ca025c [] [History] INFO  Catchup mode 'complete', applying ledger 240767/337983, queued checkpoints: 219
```

**Comment by graydon** [comment]
The "why" here is that it's bouncing off the scheduler for each ledger in the checkpoint, and is not-entirely-obviously logging the ledger number of the checkpoint each time around (there are 64 ledgers in a checkpoint, so you'll see the same log-line 64 times).

I can reword this to say "checkpoint ..." or try to thread the current ledger number through to the log line or we can just ignore it for now.

**Comment by graydon** [comment]
This behavior was fixed during the rewrite in #951

**Comment by graydon** [comment]
This behavior was fixed during the rewrite in #951
